eth/backend.go

```diff
@@ -231,6 +232,15 @@ func New(ctx *node.ServiceContext, config *Config) (*Ethereum, error) {
 		return nil, err
 	}
 
+	// Check for invalid shutdown
+	invalidShutdown, _ := chainDb.Get([]byte("unsafe-shutdown"))
```

Comment:
Since you're using `unsafe-shutdown` in quite a few places and in different packages, it might make sense to move this into `core/rawdb/schema.go`. We have a lot of other metadata keys defined there.